api/org/commcare/api/json/JsonActionUtils.java

```diff
@@ -0,0 +1,262 @@
+package org.commcare.api.json;
+
+import org.commcare.api.util.ApiConstants;
+import org.commcare.modern.util.Pair;
+import org.javarosa.core.model.Constants;
+import org.javarosa.core.model.FormDef;
+import org.javarosa.core.model.FormIndex;
+import org.javarosa.core.model.SelectChoice;
+import org.javarosa.core.model.data.*;
+import org.javarosa.core.model.data.helper.Selection;
+import org.javarosa.core.model.instance.TreeReference;
+import org.javarosa.form.api.FormEntryController;
+import org.javarosa.form.api.FormEntryModel;
+import org.javarosa.form.api.FormEntryPrompt;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.util.*;
+
+/**
+ * Utility functions for performing some action on a Form and receiving a JSON response
+ */
+public class JsonActionUtils {
+
+    /**
+     * Delete a repeat at the specified index, return the JSON response
+     *
+     * @param controller      the FormEntryController under consideration
+     * @param model           the FormEntryModel under consideration
+     * @param formIndexString the form index of the repeat group to be deleted
+     * @return The JSON representation of the updated form tree
+     */
+    public static JSONObject deleteRepeatToJson(FormEntryController controller,
+                                                FormEntryModel model, String formIndexString) {
+        JSONObject ret = new JSONObject();
+        FormIndex formIndex = indexFromString(formIndexString, model.getForm());
+        controller.deleteRepeat(formIndex);
+        ret.put(ApiConstants.QUESTION_TREE_KEY, walkToJSON(model, controller));
```

Comment:
Could return result of `getCurrentJson` to avoid repeating this line (same for `descendRepeatToJson`)